.github/workflows/ci.yaml

```diff
@@ -0,0 +1,514 @@
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-build
+
+      - name: Unpack scanner build
+        run: |
+          tar xvzf scanner-build.tgz
+
+      - name: Run db integration tests
+        run: ./scripts/ci/jobs/db-integration-tests.sh
+
+  generate-genesis-dump:
+    if: |
+      github.event_name != 'pull_request' ||
+      contains(github.event.pull_request.labels.*.name, 'generate-dumps-on-pr')
+    env:
+      NVD_API_KEY: ${{ secrets.NVD_API_KEY }}
+    runs-on: ubuntu-latest
+    needs:
+      - pre-build-updater
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: updater-build
+
+      - name: Unpack updater build
+        run: |
+          tar xvzf updater-build.tgz
+
+      - name: genesis-dump
+        run: |
+          source ./scripts/ci/lib.sh
+          generate_genesis_dump
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: genesis-dump
+          path: /tmp/genesis-dump/genesis-dump.zip
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: vuln-dump
+          path: /tmp/vuln-dump
+
+  generate-db-dump:
+    if: |
+      github.event_name != 'pull_request' ||
+      contains(github.event.pull_request.labels.*.name, 'generate-dumps-on-pr')
+    runs-on: ubuntu-latest
+    needs:
+      - generate-genesis-dump
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: updater-build
+
+      - name: Unpack updater build
+        run: |
+          tar xvzf updater-build.tgz
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: genesis-dump
+          path: /tmp/genesis-dump
+
+      - name: db-dump
+        run: |
+          source ./scripts/ci/lib.sh
+          generate_db_dump
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: db-dump
+          path: /tmp/postgres/pg-definitions.sql.gz
+
+  generate-scanner-bundle:
+    runs-on: ubuntu-latest
+    needs:
+      - pre-build-scanner
+      - generate-genesis-dump
+    if: |
+      always() &&
+      (needs.generate-genesis-dump.result == 'success' || needs.generate-genesis-dump.result == 'skipped')
```

Comment:
when is `always()` false?

Reply:
`always()` will never be false, but I think I know what you're getting at. Here's what happens when you don't have `always()`: https://github.com/stackrox/scanner/actions/runs/7468175561

My understanding from the [documentation](https://docs.github.com/en/actions/learn-github-actions/expressions#always) is the condition won't even execute unless `always()` is present